unix/x0vncserver/x0vncserver.cxx

```diff
@@ -300,7 +319,9 @@ int main(int argc, char** argv)
 
     VNCServerST server(desktopName, &desktop);
 
-    if (rfbunixpath.getValueStr()[0] != '\0') {
+    if (createSystemdListeners(&listeners)) {
+      vlog.info("Listening on systemd sockets");
+    } else if (rfbunixpath.getValueStr()[0] != '\0') {
```

Comment:
The `else` seems improper here. Let's respect all the options given.

Reply:
When using systemd, ALL listening sockets will be passed through filedescriptors. The server shouldn't attempt to open any further sockets on its own. So this flow is on purpose. Otherwise you'd be obligated to explicitly disable the other (default) sockets.

Reply:
There were some changes between this version and the one I tested with, I've updated the code to handle things as intended.